README.md

```diff
@@ -1,50 +1,37 @@
-DNS query functions
-===================
+## Table of Contents
 
-This module contains query functions for DNS for use from Puppet.
+1. [Overview](#overview)
+1. [Setup - The basics of getting started with dnsquery](#setup)
+1. [Usage - Configuration options and additional functionality](#usage)
+1. [Development - Guide for contributing to the module](#development)
 
-Usage
------
+## Overview
 
-All functions can take a optional second lambda argument that is called if the lookup returned no values. Then the result of the lambda will be used instead.
+This module provides functions and facts for querying dns records
 
-### dns_lookup
+## Setup
 
-Does a DNS lookup and returns an array of addresses.
+```console
+ % puppet module install puppet-dnsquery
+```
 
-### dns_a
+## Usage
 
-Retrieves DNS A records and returns it as an array. Each record in the
-array will be a IPv4 address.
+Since `puppet/dnsquery` version 4, all functions are provided under the `dnsquery` namespace.
+Non namespaced versions will be removed in version 5
 
-### dns_aaaa
+All functions are documented in [REFERENCE.md](REFERENCE.md)
 
-Retrieves DNS AAAA records and returns it as an array. Each record in the
-array will be a IPv6 address.
+## Development
 
-### dns_cname
+We highly welcome new contributions to this module, especially those that
+include documentation, and rspec tests ;) but will happily guide you through
+the process, so, yes, please submit that pull request!
 
-Retrieves a DNS CNAME record and returns it as a string.
+Reference documentation is generated using puppet-strings.
+To regenerate it, please run the rake task as follows.
 
-### dns_mx
+```console
+bundle exec rake reference
```

Comment:
We want to deprecate this task, in favor of using `strings:generate:reference`. Though I think we should refer to `CONTRIBUTING.md`. If that doesn't cover it, then it's something we should fix instead.

Reply:
i updated to use strings:generate:reference.  this repo doesn't have a CONTRIBUTING.md i checked a few other voxpupil projects and couldn't  quickly find one to copy.  can you send a link to one i could add.

Reply:
We store it in `.github`: https://github.com/voxpupuli/puppet-dnsquery/blob/master/.github/CONTRIBUTING.md (comes from https://github.com/voxpupuli/modulesync_config/blob/master/moduleroot/.github/CONTRIBUTING.md.erb)

Reply:
ahh thanks, added a link